### DISCUSSION TITLE
Trouble with keymaps of `copilot`

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#562] [Q&A]
I have enabled `copilot-lua` by following the extra feature:

```lua
{ import = "lazyvim.plugins.extras.coding.copilot" },
```
But the default keymaps for `prev` (`<M-[>`) and `next` (`<M-]>`) do not work well. According to [copilot.lua](https://github.com/zbirenbaum/copilot.lua),

> When auto_trigger is false, use the next or prev keymap to trigger copilot suggestion.

It seems that `M` has been mapped to `Esc` in lazyvim, so typing `M-]` in *insert* mode will be mapped to `]` in *normal* mode. How can I make `prev` (`<M-[>`) and `next` (`<M-]>`) behave as expected? 

----
Is it caused by my terminal settings? I am using `Konsole` in Manjaro.

**Comment by Jomik** [comment]
Isn't this because we use [copilot-cmp.lua](https://github.com/zbirenbaum/copilot-cmp) ?
The default suggestion is disabled https://github.com/LazyVim/LazyVim/blob/7ce1b1519a45f48aa4b5f35b7b7a886b42fe56d8/lua/lazyvim/plugins/extras/coding/copilot.lua#L9

**Comment by ChenZhongPu** [comment]
I did two more thing to make it work.

1. Specify the key-mapping in config.
2. Remap `option` key in the terminal. For example, make left option as *meta* in iterm2; set `left_option_as_alt` in Kitty.

**Comment by tienlonghungson** [comment]
@ChenZhongPu  I also have the same problem. Did you figure it out?

**Reply by ChenZhongPu** [reply]
Here is my workaround way to use `prec` and `next` by remapping them in the *insert* mode:

```lua
vim.keymap.set("i", "<M-[>", function()
  require("copilot.suggestion").prev()
end)

vim.keymap.set("i", "<M-]>", function()
  require("copilot.suggestion").next()
end)

vim.keymap.set("i", "<M-a>", function()
  require("copilot.suggestion").accept()
end)

vim.keymap.set("i", "<M-d>", function()
  require("copilot.suggestion").dismiss()
end)
```

Note that in this case, you should use `<M-a>` to accept the AI code.

Plz let me know if you have better solutions.

**Reply by tienlonghungson** [reply]
Thanks. I don't know a better solution.
